pkg/plugin/bundle.go

```diff
@@ -96,23 +103,17 @@ func NewBundle(name string, version Version, deprecateWarning string, plugins ..
 
 // NewBundleWithOptions creates a new Bundle with the provided BundleOptions.
 // The list of supported project versions is computed from the provided plugins in options.
-func NewBundleWithOptions(opts ...BundleOption) (Bundle, error) {
-	bundleOpts := bundle{}
-
-	for _, opts := range opts {
-		opts(&bundleOpts)
-	}
-
-	supportedProjectVersions := CommonSupportedProjectVersions(bundleOpts.plugins...)
+func (bp *BundlePlugin) NewBundleWithOptions() (Bundle, error) {
+	supportedProjectVersions := CommonSupportedProjectVersions(bp.Options.plugins...)
 	if len(supportedProjectVersions) == 0 {
 		return nil, fmt.Errorf("in order to bundle plugins, they must all support at least one common project version")
 	}
 
 	// Plugins may be bundles themselves, so unbundle here
 	// NOTE(Adirio): unbundling here ensures that Bundle.Plugin always returns a flat list of Plugins instead of also
 	//               including Bundles, and therefore we don't have to use a recursive algorithm when resolving.
-	allPlugins := make([]Plugin, 0, len(bundleOpts.plugins))
-	for _, plugin := range bundleOpts.plugins {
+	allPlugins := make([]Plugin, 0, len(bp.Options.plugins))
+	for _, plugin := range bp.Options.plugins {
```

Comment:
Was there a specific reason to make a nested struct?

Reply:
@varshaprasad96 Sorry about that. Changed it.

Actually, according to me the changes I made in this https://github.com/kubernetes-sigs/kubebuilder/pull/3312 for https://github.com/kubernetes-sigs/kubebuilder/issues/3307 were reasonable, since it implemented the functional arguments for the `NewBundleWithOptions` func for creating Bundles to prevent the long function signature for the [NewBundle](https://github.com/kubernetes-sigs/kubebuilder/blob/0ceec7702d6fa6f229e414dc3d9af861a75d6a59/pkg/plugin/bundle.go#L77) func.

But @camilamacedo86 has suggested that using struct for bundle options is more reasonable. 

But with the calling the function for bundle creation becomes very long like https://github.com/kubernetes-sigs/kubebuilder/blob/0ceec7702d6fa6f229e414dc3d9af861a75d6a59/cmd/main.go#L50 Which also makes the lint test fails to have to skip them, therefore https://github.com/kubernetes-sigs/kubebuilder/pull/3379#discussion_r1181602409.

Please inform me if my changes are reasonable and why this approach is better than https://github.com/kubernetes-sigs/kubebuilder/pull/3312. Maybe I am missing here something.